docs/en/reference/events.rst

```diff
@@ -164,7 +164,8 @@ the life-time of their registered entities.
    database insert operations. Generated primary key values are
    available in the postPersist event.
 -  preUpdate - The preUpdate event occurs before the database
-   update operations to entity data. It is not called for a DQL UPDATE statement.
+   update operations to entity data. It is not called for a DQL UPDATE statement
+   nor when the computed changeset is empty.
```

Comment:
>empty changeset === no update

Maybe I misunderstand you but that is what I mean. This behaviour is not documented as far as I can tell.

This is the only precondition I can think of right now that seem to be undocumented but maybe there are others.